snakebite/client.py

```diff
@@ -1412,6 +1440,29 @@ def _switch_namenode(self, namenodes):
             msg += "\nLook into debug messages - add -D flag!"
             raise OutOfNNException(msg)
 
+    def __calculate_exponential_time(self, time, retries, cap):
+        # Same calculation as the original Hadoop client but converted to seconds
+        baseTime = min(time * (1L << retries), cap);
+        return (baseTime * (random.random() + 0.5)) / 1000;
+
+    def __do_retry_sleep(self, retries):
+        # Don't wait for the first retry.
+        if (retries == 0):
+            return
+        else:
+            sleep_time = self.__calculate_exponential_time(self.base_sleep, retries, self.max_sleep)
+            log.debug("Doing retry sleep for %s seconds", sleep_time)
+            time.sleep(sleep_time)
+
+    def __should_retry(self):
+        if self.retries >= self.max_retries:
+            return False
+        else:
+            self.__do_retry_sleep(self.retries)
+            log.debug("Running retry %d of %d", self.retries, self.max_retries)
+            self.retries += 1
+            return True
+
     def __handle_request_error(self, exception):
```

Comment:
There is actually a RetriableException exception that wraps other exceptions and in this case one might retry. However it's barely used. But I'll have a look and try to get it in.